examples/Example/src/new_api/calculator/index.tsx

```diff
@@ -0,0 +1,399 @@
+  return (
+    <View style={styles.expression}>
+      <View style={{ flexDirection: 'column' }}>
+        <Text style={styles.expressionText}>{props.expression}</Text>
+        <Text style={styles.expressionResult}>{props.expression}</Text>
+      </View>
+    </View>
+  );
+}
+
+function Input(props: {
+  setHistory: Dispatch<SetStateAction<string[]>>;
+  setExpression: Dispatch<SetStateAction<string>>;
+  measure: (e: LayoutChangeEvent) => void;
+  offset: Animated.SharedValue<number>;
+  expression: string;
+}) {
+  const translationStyle = useAnimatedStyle(() => {
+    return {
+      transform: [{ translateY: props.offset.value }],
+    };
+  });
+
+  function append(symbol: string) {
+    if (symbol === '<') {
+      props.setHistory((h) => [...h, props.expression]);
+      props.setExpression((_e) => '');
+    } else {
+      props.setExpression((e) => e + symbol);
+    }
+  }
+
+  return (
+    <Animated.View
+      style={[styles.input, translationStyle]}
+      onLayout={props.measure}>
+      <NumPad append={append} />
+      <Operations />
+    </Animated.View>
+  );
+}
+
+function NumPad(props: { append: (text: string) => void }) {
+  const buttons = ['7', '8', '9', '4', '5', '6', '1', '2', '3', '<', '0', '.'];
+  return (
+    <View style={styles.numPad}>
+      {buttons.map((text) => {
+        return <Button text={text} key={text} append={props.append} />;
+      })}
+    </View>
+  );
+}
+
+function Operations() {
+  const layout = useSharedValue<LayoutRectangle>({
+    x: 0,
+    y: 0,
+    width: 0,
+    height: 0,
+  });
+  const drag = useSharedValue(0);
+  const dragOffset = useSharedValue(0);
+  const [opened, setOpened] = useState(false);
+
+  function open() {
+    const margin = window.width - layout.value.x;
+
+    drag.value = withTiming(-layout.value.width + margin, {
+      duration: DRAG_ANIMATION_DURATION,
+    });
+    dragOffset.value = -layout.value.width + margin;
+
+    setOpened(true);
+  }
+
+  function close() {
+    drag.value = withTiming(0, { duration: DRAG_ANIMATION_DURATION });
+    dragOffset.value = 0;
+
+    setOpened(false);
+  }
+
+  const dragGesture = useAnimatedGesture(
+    Gesture.pan()
+      .setOnUpdate((e) => {
+        'worklet';
+        const margin = window.width - layout.value.x;
+        const value = dragOffset.value + e.translationX;
+
+        if (value < -layout.value.width + margin) {
+          drag.value = -layout.value.width + margin;
+        } else if (value > 0) {
+          drag.value = 0;
+        } else {
+          drag.value = value;
+        }
+      })
+      .setOnEnd((e) => {
+        'worklet';
+        const margin = window.width - layout.value.x;
+        const value = dragOffset.value + e.translationX;
+
+        if (opened) {
+          if (value > -layout.value.width + margin + 75) {
+            runOnJS(close)();
+          } else {
+            runOnJS(open)();
+          }
+        } else {
+          if (value < -75) {
```

Comment:
Extract it to constant.

Reply:
Done in 26c0974724e12b9c9e06ce531674471ac46d05c3